distributed/protocol/core.py

```diff
@@ -105,8 +145,15 @@ def _decode_default(obj):
                     )
                 else:
                     return Serialized(sub_header, sub_frames)
-            else:
-                return msgpack_decode_default(obj)
+
+            offset = obj.get("__Pickled__", 0)
+            if offset > 0:
+                sub_header = msgpack.loads(frames[offset])
+                offset += 1
+                sub_frames = frames[offset : offset + sub_header["num-sub-frames"]]
+                return pickle.loads(sub_header["pickled-obj"], buffers=sub_frames)
```

Comment:
In [dask#8672](https://github.com/dask/dask/pull/8672) I am using `config.get("distributed.scheduler.pickle")` to check if the user is allowing `pickle.loads` on the scheduler. If this option is set to `False`, I am converting every layer to a `MaterializedLayer` before shipping to the scheduler. Perhaps we should check this config on the scheduler and throw a clear error if a `"__Pickled__"` object is detected?

Do you have other thoughts on what to do when the "pickle" option is False?

My suggestion assumes that we will still want to preserve annotations when the scheduler is in "restricted" mode (rather than just materializing the entire graph).

Reply:
> Do you have other thoughts on what to do when the "pickle" option is False?

Nope, I totally agree; the scheduler should raise an error.